src/test/java/com/amazon/ion/GoodIonMultithreadedTest.java

```diff
@@ -0,0 +1,181 @@
+package com.amazon.ion;
+
+import com.amazon.ion.junit.Injected;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+
+import static com.amazon.ion.TestUtils.GLOBAL_SKIP_LIST;
+import static com.amazon.ion.TestUtils.GOOD_IONTESTS_FILES;
+import static com.amazon.ion.TestUtils.testdataFiles;
+
+/**
+ * Tests IonValue's multithreaded contract over the `good` ion-tests files.
```

Comment:
```suggestion
 * Tests IonValue's multithreaded contract over the `good` ion-tests files.
 *
 * For each file, the test randomly chooses a child value to test and marks that value as read-only. Then
 * it spawns many instances of an async task to concurrently perform read operations on the chosen value.
 * Finally, it asserts that all instances of the async task got the same results from the read operations.
 *
 * Testing every value at every depth would be more robust, but is likely more suited to a suite of 
 * long-running tests that are executed periodically, rather than the unit tests that are executed on each build.
 * This test is expected to take only a few seconds to execute.
```